### DISCUSSION TITLE
Benchmark App Freeze For Long Videos

### DISCUSSION THREAD

**Discussion by GilgusMaximus** [FreeTubeApp/FreeTube#3622] [General]
**Topic**
- [x] App performance

**Discussion**
Hello everyone. After #1065 presented the case that the application seems to freeze on his system, even after the video below starts playing in DASH format, I want you guys to helps me out here and benchmark how long the application freezes for you.  Additionally it is important to check whether you can use the player controls like fullscreen, playback speed, fullwindow etc.. normally after the video starts playing. The reasoning behind this is that for me the app loads rather quickly and then works just fine, while @ModdedGamers seems to have problems with that. So I just want to get a view on how many people have such a problem for longer videos.

The video:
https://youtu.be/_rnUCkKjJZQ

So the steps for the benchmark would be (with DASH as default method of video playback):
1) Copy the video link.
2) Paste it into the search and search for it to open the video
3) Let the video load and test whether everything works and check roughly how long it took until the app unfroze
4) Share your results in this thread

Thanks in advance for the help !

Best regards,
Gilgus

**Comment by pyrox0** [comment]
As the person who originated the issue, here are my findings:

* 5 seconds to unfreeze
* Since the menu would not open, I would estimate the video to be in either 360p or 480p
* I clicked the pause button as soon as the video loaded, and it took 10 seconds to pause, as indicated by the video timer, which also did not unfreeze until the video was paused.

In case it's important. I am running this on a 2017 Macbook Air.

**Comment by nodeg** [comment]
- After 5 seconds there was a circle which indicated the video was loading
- After 9 seconds the video start playing
- As soon as the video played I started to pause it. It stopped after 12 s accoding to the video timer.
- FreeTube behaves normally after I paused the video. The player controls work.

Version: v0.11.3 Beta, v0.12.0 Beta
OS: macOS Big Sur 11.2.2 on a MacBook Pro 2020, openSUSE Tumbleweed on a Dell Latitude 5200

**Comment by My-Random-Thoughts** [comment]
took about 14ish seconds to start from clicking play.  Everything worked normally with no issues.  Speed up and slowdown worked, play/pause worked

Version 0.12.0beta
OS: Ubuntu Linux

**Comment by lfom** [comment]
I think there are two different problems, maybe one with the page itself and another with the video. I have auto-play disabled, so when I open the video page it stutters while scrolling for about 10s, and then I the video starts to play right after I click play, but then FreeTube stutters for about 10s more. If I try to play the video right after its page is opened then it is much worse: the video takes about 20s to start playing (and it plays in lowres) and then the whole page stutters for about 10s more. After that, in both cases, zoom, fullscreen, speed, etc behave normally.

v.0.12.0 Beta
Pop!_OS 20.10 (like Ubuntu 20.10)
i5-2500S 12GB ATI 6770m

**Comment by computeristic** [comment]
The video opened as soon as I searched for the link. While it was loading, the app froze for 35 seconds (couldn't scroll, or press anything), then the video started playing and everything got back to normal (can pause, skip, scroll, fullscreen, and navigate the app). 

FT version: v0.13.1 Beta
OS: macOS Mojave on a macbook pro 2014

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
30seconds to unfreeze while it was frozen i couldn't scroll, or press anything, then the video started playing and all the functions of the player worked and navigating the app also worked.
